pkg/liquidity-mining/contracts/fee-distribution/FeeDistributorBALClaimer.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-standalone-utils/contracts/interfaces/IBALTokenHolder.sol";
+
+import "../interfaces/IFeeDistributor.sol";
+import "../interfaces/ISingleRecipientGauge.sol";
+
+/**
+ * @title FeeDistributorBALClaimer
+ * @notice Atomically mints any outstanding BAL from a SingleRecipientGauge and transfers it to the FeeDistributor
+ * in order for it to be distributed among veBAL holders.
+ */
+contract FeeDistributorBALClaimer {
+    IERC20 private immutable _balToken;
+    IFeeDistributor private immutable _feeDistributor;
+    ISingleRecipientGauge private immutable _gauge;
+    IBALTokenHolder private immutable _balTokenHolder;
+
+    constructor(IFeeDistributor feeDistributor, ISingleRecipientGauge gauge) {
+        IBALTokenHolder balTokenHolder = IBALTokenHolder(gauge.getRecipient());
+
+        _balToken = balTokenHolder.getBalancerToken();
+        _feeDistributor = feeDistributor;
+        _gauge = gauge;
+        _balTokenHolder = balTokenHolder;
+    }
+
+    /**
+     * @notice Mint any outstanding BAL emissions and send them to the FeeDistributor
```

Comment:
Maybe add a note that it needs permissions to withdraw funds from BALTokenHolder